app/controllers/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+class UsersController < ApplicationController
+  def show
+    find_user
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to user_path(@user)
+    else
+      respond_with(@user)
+    end
+  end
+
+  def edit
+    find_user
+  end
+
+  def update
+    find_user
+    if @user.update(user_params)
+      redirect_to user_path(@user)
+    else
+      respond_with(@user)
+    end
+  end
+
+  def destroy
+    find_user.delete
+    redirect_to root_path
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password, :uid)
+  end
+
+  def find_user
+    @user = User.find(params[:id])
+  end
```

Comment:
You can also add:
`helper_method :user`
And then just use that in the views, which means you could then remove the call to find user in the show and edit actions

Reply:
Mmm, ok will investigate this, sounds good, thanks Elle